docs/proposals/cascading-deletion.md

```diff
@@ -85,11 +85,13 @@ The only new component is the Garbage Collector, which consists of a scanner, a
 * Garbage Processor:
   * Consists of the *Dirty Queue* and workers.
   * Each worker:
-    * Dequeues an item from *Dirty Queue*
-    * Checks if the item’s parents still exist.
-    * Checks for a tombstone of parents that don’t exist
-    * If there is no parent nor tombstone, requests the API server to delete the item.
-    * If tombstone does exist, removes the parent from the object’s ParentReferences.
+    * Dequeues an item from *Dirty Queue*.
+    * If the item's ParentReferences is empty, continues to process the next item in the *Dirty Queue*.
+    * Otherwise checks each entry in the ParentReferences:
+      * If a parent exists, continues to check the next parent.
```

Comment:
If a parent exists, the item will not be deleted this iteration.

Do we need to still check all the parents anyway?